embedchain/loaders/site_map.py

```diff
@@ -0,0 +1,33 @@
+import requests
+from bs4 import BeautifulSoup
+
+from embedchain.loaders.web_page import WebPageLoader
+
+
+class SitemapLoader:
+    def load_data(self, sitemap_url):
+        """
+        This method takes a sitemap url as input and retrieves
+        all the urls to use the WebPageLoader to load content
+        of each page.
+        """
+        output = []
+        web_page_loader = WebPageLoader()
+
+        response = requests.get(sitemap_url)
+
+        if response.status_code == 200:
+            soup = BeautifulSoup(response.text, features="xml")
+            links = [link.text for link in soup.find_all("loc")]
+
+            for link in links:
+                each_load_data = web_page_loader.load_data(link)
+                # WebPageLoader returns a list with single element
+                # which is extracted and appended to the output list
+                # containing data for all pages
+                output.append(each_load_data[0])
+
+            return output
+
+        else:
+            raise response.raise_for_status()
```

Comment:
You can make the code a lot simpler by doing something like this:

```python
import requests
from bs4 import BeautifulSoup
from embedchain.loaders.web_page import WebPageLoader


class SitemapLoader:
    def load_data(self, sitemap_url):
        """
        This method takes a sitemap URL as input and retrieves
        all the URLs to use the WebPageLoader to load content
        of each page.
        """
        output = []
        web_page_loader = WebPageLoader()
        response = requests.get(sitemap_url)
        response.raise_for_status()

        soup = BeautifulSoup(response.text, "xml")
        links = [link.text for link in soup.find_all("loc")]
        for link in links:
            each_load_data = web_page_loader.load_data(link)
            output.append(each_load_data)
        return [data[0] for data in output]
```

Make sure to run `make format` after making the change.

Reply:
updated @deshraj